exercises/list-ops/test/test_list_ops.c

```diff
@@ -0,0 +1,325 @@
+#include "vendor/unity.h"
+#include "../src/list_ops.h"
+
+void setUp(void)
+{
+}
+
+void tearDown(void)
+{
+}
+
+static void check_lists_match(list_t * expected, list_t * actual)
+{
+   // check inputs to be valid lists
+   TEST_ASSERT_NOT_NULL(expected);
+   TEST_ASSERT_NOT_NULL(actual);
+
+   // check lengths match
+   TEST_ASSERT_EQUAL_MESSAGE(expected->length, actual->length,
+                             "List lengths differ");
+
+   // check values match in non-zero length lists
+   if (expected->length)
+      TEST_ASSERT_EQUAL_MEMORY_ARRAY(expected, actual, sizeof(list_value_t),
+                                     expected->length);
+}
+
+static bool filter_modulo(list_value_t value)
+{
+   return (value % 2 == 1);
+}
+
+static list_value_t map_increment(list_value_t value)
+{
+   return value + 1;
+}
+
+static list_value_t fold_multiply(list_value_t value, list_value_t initial)
+{
+   return value * initial;
+}
+
+static list_value_t fold_add(list_value_t value, list_value_t initial)
+{
+   return value + initial;
+}
+
+static list_value_t fold_divide(list_value_t value, list_value_t initial)
+{
+   return (initial == 0) ? 0 : value / initial;
+}
+
+static void test_append_empty_lists(void)
+{
+   list_t *list1 = new_list(0, NULL);
+   list_t *list2 = new_list(0, NULL);
+   list_t *expected = new_list(0, NULL);
+
+   list_t *actual = append_list(list1, list2);
+   check_lists_match(expected, list1);
+
+   delete_list(list1);
+   delete_list(list2);
+   delete_list(expected);
+   delete_list(actual);
+}
+
+static void test_append_list_to_empty_list(void)
+{
+   TEST_IGNORE();               // delete this line to run test
+   list_t *list1 = new_list(0, NULL);
+   list_t *list2 = new_list(3, (list_value_t[]){ 1, 3, 4 });
+   list_t *expected = new_list(3, (list_value_t[]){ 1, 3, 4 });
+
+   list_t *actual = append_list(list1, list2);
+   check_lists_match(expected, actual);
+
+   delete_list(list1);
+   delete_list(list2);
+   delete_list(expected);
+   delete_list(actual);
+}
+
+static void test_append_empty_list_to_list(void)
+{
+   TEST_IGNORE();
+   list_t *list1 = new_list(4, (list_value_t[]){ 1, 2, 3, 4 });
+   list_t *list2 = new_list(0, NULL);
+   list_t *expected = new_list(4, (list_value_t[]){ 1, 2, 3, 4 });
+
+   list_t *actual = append_list(list1, list2);
+   check_lists_match(expected, actual);
+
+   delete_list(list1);
+   delete_list(list2);
+   delete_list(expected);
+   delete_list(actual);
+}
+
+static void test_append_non_empty_lists(void)
+{
+   TEST_IGNORE();
+   list_t *list1 = new_list(2, (list_value_t[]){ 1, 2 });
```

Comment:
We probably don't want to blow people's minds, but it would be cool if you could specify a new list using variadic arguments

Reply:
Coded this up out of curiosity because I realised I have somehow never used variadic arguments in C.
While it results in a nicer interface, it feels ... janky.  The type (un)safety 😱

In the main I think not a good idea though as this exercise already introduces one other new thing: function pointers  🙌 . 
Neither function pointers nor variadic functions are covered by any other exercise yet. While the exercise can work without variadic functions, I think it cannot work without function pointers.